File: 02Fabryka/Students/2017/AbstractFactory/src/design/patterns/books/Book.java
package design.patterns.books;

import design.patterns.books.enums.Binding;
import design.patterns.books.enums.BookType;
import design.patterns.parts.authors.Authors;
import design.patterns.parts.publisher.Publisher;

public abstract class Book {

    protected BookType bookType;
    protected Integer pages;
    protected Authors author;
    protected Publisher publisher;

    public abstract void prepare();

    public void collecting(){
        System.out.println("Collecting order");
    }
    public void packing(){
        System.out.println("Packing order");
    }

    public void send(){
        System.out.println("Sending book");
    }

    @Override
    public String toString() {
        return "Book{" +
                "bookType=" + bookType +
                ", pages=" + pages +
                ", author='" + author + '\'' +
                ", publisher='" + publisher + '\'' +
                '}';
    }
}

File: 02Fabryka/Students/2018/OlszerKrzysztof/Factory/FactoryMethod/Books/Book.cs
namespace FactoryMethod.Books
{
    public abstract class Book
    {
        protected string Author;
        protected string Title;
        protected int NoPages;

        public string open() => "Book.open()";
        public string getName() => Author + " " + Title;
        public int getNoPages() => NoPages;
    }
}
